lib/Dialect/Secret/IR/SecretOps.cpp

```diff
@@ -20,6 +20,24 @@ ParseResult YieldOp::parse(OpAsmParser &parser, OperationState &result) {
                  parser.resolveOperands(opInfo, types, loc, result.operands));
 }
 
+LogicalResult YieldOp::verify() {
+  // Trait verifier ensures parent is a GenericOp
+  auto parent = llvm::cast<GenericOp>(getParentOp());
+
+  for (int i = 0; i < getValues().size(); ++i) {
+    auto yieldSecretType = SecretType::get(getValues().getTypes()[i]);
+    if (yieldSecretType != parent.getResultTypes()[i]) {
+      return emitOpError()
+             << "Result types of yield op must match result types of enclosing "
+                "generic op. Yield op result types: "
+             << getValues().getTypes()
+             << "; enclosing generic op result types: "
+             << parent.getResultTypes();
```

Comment:
Should this be parent.getResultTypes()[i] ?

(Im not entirely familiar with how MLIR error messages work, but I would assume it will throw some form of array indexing error)

Reply:
I could just print one type, but I think it would be more helpful to show all the result types here. e.g., if the yield takes as input i32, i32, and the generic has results secret.secret<i32> and secret.secret<i64> , it might be confusing somehow if you're looking at the first argument and not realizing it's reporting an error about the second argument. But if you see the whole thing then it might be easier to read.

Reply:
Ah got it! Yes whole thing is easier.

My bad! i assumed the iterator would throw some form of error, but I tried it and it prints just fine (getting used to MLIR)